packages/affiliates/libs/contracts.js

```diff
@@ -25,14 +31,51 @@ function DecodeTransaction(abi, meta = {}) {
   };
 }
 
-function decodeAttribution(
-  data,
-  delimiter = "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff000000000000000000000000"
-) {
-  assert(data, "requires data to decode");
-  return data.split(delimiter)[1];
+function GetInputLength(abi) {
+  // returns length in bits of solidity type. not exhaustive but covers common params.
+  // see more: https://docs.soliditylang.org/en/v0.5.3/abi-spec.html#argument-encoding
+  function typeToLength(type) {
+    if (type === "uint") return 256;
+    if (type === "int") return 256;
+    if (type.includes("uint")) {
+      return parseInt(type.slice(4));
+    }
+    if (type.includes("int")) {
+      return parseInt(type.slice(3));
+    }
+    if (type.includes("bool")) return 8;
+    if (type == "address") {
+      return 160;
+    }
+    throw new Error("Unknown type specified: " + type);
+  }
+  function componentLength(component) {
+    return typeToLength(component.type);
+  }
+  function componentsLength(components = []) {
+    return components.reduce((sum, component) => {
+      return sum + componentLength(component);
+    }, 0);
+  }
+  return name => {
+    const find = abi.find(x => x.name === name);
+    assert(find, "unable to find name in abi: " + name);
+    if (find.inputs == null || find.inputs.length == 0) return 0;
+    return find.inputs.reduce((length, input) => {
+      return length + componentsLength(input.components);
+    }, 32); // 4 bytes always added as function name header hash
+  };
 }
 
+const DecodeAttribution = (abi, defaultAddress) => {
+  const inputLength = GetInputLength(abi)("create") / 4 + 2;
```

Comment:
can you add a comment on why you divide by 4 and add 2 to compute the `inputLength`? 
```suggestion
  const inputLength = GetInputLength(abi)("create") / 4 + 2; // convert bits to hex and + 2 for 0x
```